nemo/collections/llm/gpt/model/base.py

```diff
@@ -157,7 +157,9 @@ def forward_step(self, batch) -> torch.Tensor:
     def training_step(self, batch, batch_idx=None) -> torch.Tensor:
         # In mcore the loss-function is part of the forward-pass (when labels are provided)
 
-        return self.forward_step(batch)
+        result = self.forward_step(batch)
+        self.log('step', torch.tensor(self.trainer.global_step, dtype=torch.float32))
```

Comment:
Can't we put this inside the `training_step` of the `MegatronStrategy`?

Reply:
That way we can do it once as opposed to each individual model having to do this. The `MegatronStrategy` will wrap the `training_step` here